File: 프로그래머스/lv2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int length = people.length;

        Arrays.sort(people);
        int index = 0;
        for (int i = length - 1; i >= index; i--) {
            if (people[i] + people[index] <= limit) {
                index++;
                answer++;
            } else {
                answer++;
            }

        }
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => a - b);
    let lt = 0;
    for(let rt = people.length - 1; lt < rt; rt--) {
        if( people[lt] + people[rt] <= limit ) lt++;
    }    
    return people.length - lt;
}